packages/shared/src/events/generic-event-emitter.ts

```diff
@@ -29,15 +34,16 @@ export abstract class GenericEventEmitter<E extends AnyProviderEvent, Additional
       await handler(details);    
     };
     // The async handler has to be written to the map, because we need to get the wrapper function when deleting a listener
-    this._handlers.set(handler, asyncHandler);
```

Comment:
If the same handler (let's say 0x00000001) is added twice, the first time it's added to the weakmap by it's reference with a new wrapped async ref as a value (`0x00000001`:`0x00000002`). Then, it's added again (with a new value for it's new async wrapper since that's a local var (`0x00000001`:`0x00000003`). Since it's the same handler being added again, it has the same key... so we've overwritten `0x00000001`:`0x00000002` with `0x00000001`:`0x00000003` and `0x00000002` is lost forever and cannot be removed :sweat_smile:

Reply:
Good catch🤦‍♂️